repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/PreProcessorUtils.java

```diff
@@ -200,4 +216,173 @@ public static void verifyDuplicateAssetByName(String typeName, String assetName,
             throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, errorMessage);
         }
     }
+
+    public static void isValidLexoRank(String entityType, String inputLexorank, String glossaryQualifiedName, String parentQualifiedName, EntityDiscoveryService discovery) throws AtlasBaseException {
+
+        Matcher matcher = LEXORANK_VALIDITY_PATTERN.matcher(inputLexorank);
+
+        if(!matcher.matches() || StringUtils.isEmpty(inputLexorank)){
+            throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, "Invalid value for lexicographicalSortOrder attribute");
+        }
+        // TODO : Need to discuss either to remove this after migration is successful on all tenants and custom-sort is successfully GA or keep it for re-balancing WF
+        Boolean requestFromMigration = RequestContext.get().getRequestContextHeaders().getOrDefault("x-atlan-request-id", "").contains("custom-sort-migration");
+        if(requestFromMigration) {
+            return;
+        }
+        Map<String, String> lexoRankCache = RequestContext.get().getLexoRankCache();
+        if(Objects.isNull(lexoRankCache)) {
+            lexoRankCache = new HashMap<>();
+        }
+        String cacheKey = entityType + "-" + glossaryQualifiedName + "-" + parentQualifiedName;
+        if(lexoRankCache.containsKey(cacheKey)){
```

Comment:
Review this.

Reply:
this is patched now
![image](https://github.com/atlanhq/atlas-metastore/assets/162408549/3618062d-602a-4cbe-8377-ab267c54cb4b)